design/protected-apps-view.md

```diff
@@ -0,0 +1,917 @@
+      kind: Placement
+      
+      # The clusters where this placement has scheduled the application
+      selectedClusters:
+        - primary-datacenter
+      
+      numberOfClusters: 1
+  
+  # Disaster recovery configuration (if protected)
+  # This is the most important section for the UI
+  drInfo:
+    enabled: true
+    
+    # Reference to the DRPC providing protection
+    drpcRef:
+      name: payment-service-drpc
+      namespace: openshift-gitops
+    
+    # The DR policy defining the DR configuration
+    drPolicyRef:
+      name: metro-dr-policy
+    
+    # The clusters involved in disaster recovery
+    drClusters:
+      - primary-datacenter
+      - secondary-datacenter
+    
+    # Current primary (active) cluster
+    primaryCluster: primary-datacenter
+    
+    # Namespaces being protected
+    protectedNamespaces:
+      - payment-service
+      - payment-service-config
+    
+    # Current DR status
+    status:
+      phase: Relocated
+      conditions:
+        - type: Available
+          status: "True"
+          lastTransitionTime: "2025-04-24T06:13:43Z"
+          reason: RecoveryCompleted
+          message: "Application successfully failed over to secondary datacenter"
+    
+    # For Subscription applications with multiple placements/DRPCs
+    additionalDRPCs:
+      - drpcRef:
+          name: payment-backend-drpc
+          namespace: openshift-gitops
+        drPolicyRef:
+          name: backup-policy
+        primaryCluster: backup-datacenter
+  
+  # Type-specific information for ApplicationSets
+  applicationSetInfo:
+    gitRepoURLs:
+      - https://github.com/example/payment-service
+    targetRevision: main
+    syncPolicy: automated
+    
+    # Sibling ApplicationSets sharing the same Placement
+    siblingApplicationSets:
+      - payment-service-canary
+
+status:
+  # Controller-maintained status
+  observedGeneration: 1
+  lastSyncTime: "2025-04-24T06:13:43Z"
+  
+  conditions:
+    - type: Ready
+      status: "True"
+      lastTransitionTime: "2025-04-24T06:13:43Z"
+      reason: FullySynchronized
+      message: "View is up-to-date with source resources"
+```
+
+The key design decisions in this CRD are worth highlighting. First, the owner reference to the source application ensures that when an ApplicationSet is deleted, its corresponding view is automatically garbage collected. This prevents orphaned view resources from accumulating.
+
+Second, the labels enable efficient queries. The frontend can watch only ProtectedApplicationViews with the label `ramendr.openshift.io/protected: "true"`, ensuring we only show protected applications. The type label allows filtering by application type without examining the spec.
+
+Third, the `drInfo.additionalDRPCs` field handles the complex case of Subscription applications with multiple placements. The primary DRPC information is in the main `drInfo` section for easy display in the table, while additional DRPCs are available for drill-down views.
```

Comment:
just asking for my ref, how do we know which one is primary and which are secondary ??

Reply:
Its just a first found selection. Primary and secondary are more like main DRPC which points to multiple subs and siblings DRPCs which point to others